packages/provider-elasticsearch/src/example-types/results/highlighting/request.js

```diff
@@ -148,7 +151,7 @@ export let getAllHighlightFields = _.memoize((schema) => {
 let collectKeysAndValues = (f, coll) =>
   F.reduceTree()(
     (acc, val, key) =>
-      f(val) ? F.push(val, acc) : f(key) ? F.push(key, acc) : acc,
+      f(val) ? F.pushOn(acc, val) : f(key) ? F.pushOn(acc, key) : acc,
```

Comment:
Same kind of thing here. Using `reduce` with a mutating call in the reducing function feels a little awkward to me. Why not just walk the tree and and `push` onto an array?